ArcadiaTeamsBot/Dialogs/OpenedRequestsDialog.cs

```diff
@@ -1,62 +1,128 @@
+        private readonly IMediator mediator;
 
-        public OpenedRequestsDialog() : base(nameof(OpenedRequestsDialog))
+        public OpenedRequestsDialog(IMediator mediator) : base(nameof(OpenedRequestsDialog))
         {
+            this.mediator = mediator;
+
             this.AddDialog(new WaterfallDialog(nameof(WaterfallDialog), new WaterfallStep[]
             {
-                TypeStep,
-                EndStep,
+                this.InfoStep,
+                this.EndStep
             }));
+
             this.AddDialog(new TextPrompt(nameof(TextPrompt)));
             this.InitialDialogId = nameof(WaterfallDialog);
         }
 
-        private static async Task<DialogTurnResult> TypeStep(WaterfallStepContext stepContext, CancellationToken cancellationToken)
+        private async Task<DialogTurnResult> InfoStep(WaterfallStepContext stepContext, CancellationToken cancellationToken)
         {
-            var attachments = new[]
-            {
-                GetInfoCard().ToAttachment()
-            };
-            return await stepContext.PromptAsync(nameof(TextPrompt),
-                new PromptOptions
-                {
-                    Prompt = (Activity)MessageFactory.Attachment(attachments),
-                }, cancellationToken);
+            var getOpenedRequestsQuery = new GetCurrentServiceDeskRequestsQuery(Username);
+            var openedRequest = await this.mediator.Send(getOpenedRequestsQuery, cancellationToken);
+
+            var actions = openedRequest.Select(GetRequestAction).ToList();
+            actions.Add(new AdaptiveSubmitAction { Title = Back, Data = Back });
+
+            await stepContext.Context.SendActivityAsync(MessageFactory.Attachment(OpenedRequestsCard(actions)), cancellationToken);
+            return await stepContext.PromptAsync(nameof(TextPrompt), new PromptOptions(), cancellationToken);
         }
 
-        private static async Task<DialogTurnResult> EndStep(WaterfallStepContext stepContext, CancellationToken cancellationToken)
+        private async Task<DialogTurnResult> EndStep(WaterfallStepContext stepContext, CancellationToken cancellationToken)
         {
-            if ((string)stepContext.Result == Back)
+            switch (stepContext.Result)
             {
-                return await stepContext.BeginDialogAsync(nameof(MainDialog), null, cancellationToken);
+                case Back:
+                    await stepContext.EndDialogAsync(null, cancellationToken);
+                    return await stepContext.BeginDialogAsync(nameof(MainDialog), null, cancellationToken);
+
+                default:
+                    return await stepContext.ReplaceDialogAsync(nameof(OpenedRequestsDialog), null, cancellationToken);
             }
-            return await stepContext.ContinueDialogAsync(cancellationToken: cancellationToken);
         }
 
-        public static HeroCard GetInfoCard()
+        private static Attachment OpenedRequestsCard(List<AdaptiveAction> actions)
```

Comment:
Method name should contain some verb like "Get".

Reply:
Renamed